src/LoRa.cpp

```diff
@@ -211,6 +212,25 @@ float LoRaClass::packetSnr()
   return ((int8_t)readRegister(REG_PKT_SNR_VALUE)) * 0.25;
 }
 
+double LoRaClass::packetFrequencyError()
+{
+  int32_t fe = (int32_t)readRegister(REG_FREQ_ERROR) & 7;
+  fe <<= 8L;
+  fe += (int32_t)readRegister(REG_FREQ_ERROR+1);
+  fe <<= 8L;
+  fe += (int32_t)readRegister(REG_FREQ_ERROR+2);
+  
+  if (readRegister(REG_FREQ_ERROR) & 8) {
+    fe = fe - 524288;
+  }
+
+  double freqError = (double)fe;
```

Comment:
Are these formula's in the data sheet somewhere?

Reply:
In the data sheet, section 4.1.5 "Frequency Error Indication"
The calculation code in https://github.com/sandeepmistry/arduino-LoRa/pull/113 looks a bit clearer. I wonder if that approach to be updating the frequency is better than just returning the frequency error value like this does?